File: C/Algorithms/Searching-Algorithms/Binary_search.c
#include<stdio.h>

#define N 10 // SIZE OF ARRAY

int low = 0, high = N-1, mid; // LOW, HIGH AND MID INDEX FOR BINARY SEARCH FUNCTION
int Index = -1; //RETURN VALUE

int Binary_search(int A[], int x) // SEARCH FUNCTION
{
	if(high > low)
	{
		mid = (low + high)/2;
		if(A[mid] == x)
			Index = mid;
		else if(A[mid] < x)
		{
			low = mid+1;
			Index = Binary_search( A, x);
		}
		else 
		{
			high = mid-1;
			Index = Binary_search( A, x);
		}
		return Index;

	}
	else
	{
		return Index;
	}
}

int main()
{
	int A[N] = {2, 5, 8 ,9 ,12, 15 ,23, 29, 35, 37}; // SORTED ARRAY
	int x = 12; // 

	int result = Binary_search( A, x);

	if(result == -1) // IF NOT FOUND
	{
		printf("%d not found\n", x);
	}
	else
	{
		printf("%d is present at index : %d\n", x, result);
	}

	return 0;


}

// OUTPUT OF PROGRAM
// 12 is present at index : 4

File: Java/Algorithms/Searching-Algorithms/binary_Search.java
import java.util.*;
class BinarySearch{  
    public static int binarySearch(int arr[], int first, int last, int key){  
        if (last>=first){  
            int mid = first + (last - first)/2;  
            if (arr[mid] == key){  
            return mid;  
            }  
            if (arr[mid] > key){  
            return binarySearch(arr, first, mid-1, key);//search in left subarray  
            }else{  
            return binarySearch(arr, mid+1, last, key);//search in right subarray  
            }  
        }  
        return -1;  
    }  
    public static void main(String args[]){
        Scanner scn=new Scanner(System.in);
        System.out.println("No. of Elements to put in array");

       int n=scn.nextInt();
        int[] arr=new int[n];
       for(int i=0;i<n;i++)
         {
           arr[i]=scn.nextInt();
         }  
          
          System.out.println("Element to Search");
        int key = scn.nextInt();  
        int last=arr.length-1;  
        int result = binarySearch(arr,0,last,key);  
        if (result == -1)  
            System.out.println("Element is not found!");  
        else  
            System.out.println("Element is found at index: "+result);  
    }  
}  